packages/circuit-ui/components/SideNavigation/types.ts

```diff
@@ -105,4 +106,8 @@ export interface SecondaryLinkProps {
    * a new link or to indicate new content.
    */
   badge?: BadgeProps;
+  /**
+   * An optional badge to highlight elements belonging to a specific tier.
+   */
+  tier?: TierIndicatorProps['variant'];
```

Comment:
This will be difficult to evolve if we want to pass additional props to the TierIndicator in the future. Let's accept the full props object here, also to match the `badge` prop above.

```suggestion
  tier?: TierIndicatorProps;
```

Reply:
for now I am forcing the size to `s`, that is why i thought it would be confusing to allow other props if they won't be reflected in the UI. I could change it to `Omit< TierIndicatorProps , 'size'>` if that's better. WDYT?